sale_promotion_rule/models/sale_promotion_rule.py

```diff
@@ -103,6 +107,49 @@ class SalePromotionRule(models.Model):
 """,
     )
 
+    count_usage = fields.Integer(
+        string="Number of use",
+        help="Total number sale order which use this rule",
+        compute="_calc_count_usage",
+    )
+    budget_spent = fields.Monetary(
+        "Budget Spent", compute="_calc_budget_spent"
+    )
+    budget_max = fields.Monetary("Budget Max")
+
+    def _calc_count_usage(self):
+        for rec in self:
+            count = self.env["sale.order"].search_count(
+                [
+                    ("state", "in", ["sale", "done"]),
+                    "|",
+                    ("promotion_rule_ids", "in", rec.ids),
+                    ("coupon_promotion_rule_id", "=", rec.id),
+                ]
+            )
+            rec.count_usage = count
+
+    def _calc_budget_spent(self):
+        for rec in self:
+            lines = self.env["sale.order.line"].search(
+                [
+                    ("state", "in", ["sale", "done"]),
+                    ("promotion_rule_id", "=", rec.id),
+                    # "|",
+                    # ("promotion_rule_ids", "in", rec.ids),
+                    # ("coupon_promotion_rule_id", "=", rec.id),
+                ]
+            )
+            if lines:
+                if rec.discount_type == "amount_tax_included":
+                    rec.budget_spent = -sum(lines.mapped("price_total"))
```

Comment:
it should not be negative

Reply:
Since it's a discount, price_total is always negative, so I negate it again to have a positive value. Maybe I should use `abs()` ?

Reply:
It's named budget_spent. So it should be positive.

Reply:
Yes, `budget_spent` is positive, this is `lines.price_total` that is negative

Reply:
Ok I get it.